Sources/X509/Verifier/OneOfPolicyBuilder.swift

```diff
@@ -0,0 +1,207 @@
+        }
+
+        @inlinable
+        var verifyingCriticalExtensions: [SwiftASN1.ASN1ObjectIdentifier] {
+            let firstExtensions = first.verifyingCriticalExtensions
+            let secondExtensions = second.verifyingCriticalExtensions
+            return firstExtensions.filter { secondExtensions.contains($0) }
+        }
+
+        @inlinable
+        mutating func chainMeetsPolicyRequirements(chain: UnverifiedCertificateChain) async -> PolicyEvaluationResult {
+            let firstResult = await self.first.chainMeetsPolicyRequirements(chain: chain)
+            switch firstResult {
+            case .meetsPolicy:
+                return .meetsPolicy
+            case .failsToMeetPolicy(let firstReason):
+                let secondResult = await self.second.chainMeetsPolicyRequirements(chain: chain)
+                switch secondResult {
+                case .meetsPolicy:
+                    return .meetsPolicy
+                case .failsToMeetPolicy(let secondReason):
+                    return .failsToMeetPolicy(reason: "\(firstReason) and \(secondReason)")
+                }
+            }
+        }
+    }
+
+    @inlinable
+    public static func buildPartialBlock<Policy: VerifierPolicy>(first: Policy) -> Policy {
+        first
+    }
+
+    @inlinable
+    public static func buildPartialBlock(
+        accumulated: some VerifierPolicy,
+        next: some VerifierPolicy
+    ) -> some VerifierPolicy {
+        Tuple2(first: accumulated, second: next)
+    }
+}
+
+// MARK: if
+extension OneOfPolicyBuilder {
+    @inlinable
+    public static func buildOptional(_ component: (some VerifierPolicy)?) -> some VerifierPolicy {
+        PolicyBuilder.WrappedOptional(component)
+    }
+}
+
+// MARK: if/else and switch
+extension OneOfPolicyBuilder {
+    @inlinable
+    public static func buildEither<First: VerifierPolicy, Second: VerifierPolicy>(
+        first component: First
+    ) -> PolicyBuilder._Either<First, Second> {
+        PolicyBuilder._Either<First, Second>(storage: .first(component))
+    }
+
+    @inlinable
+    public static func buildEither<First: VerifierPolicy, Second: VerifierPolicy>(
+        second component: Second
+    ) -> PolicyBuilder._Either<First, Second> {
+        PolicyBuilder._Either<First, Second>(storage: .second(component))
+    }
+}
+
+extension OneOfPolicyBuilder {
+    @inlinable
+    public static func buildFinalResult(_ component: some VerifierPolicy) -> some VerifierPolicy {
+        PolicyBuilder.CachedVerifyingCriticalExtensions(wrapped: component)
+    }
+
+    @inlinable
+    public static func buildFinalResult(_ component: AnyPolicy) -> AnyPolicy {
+        func unwrapExistentialAndCache(policy: some VerifierPolicy) -> some VerifierPolicy {
+            PolicyBuilder.CachedVerifyingCriticalExtensions(wrapped: policy)
+        }
+        let cachedPolicy = unwrapExistentialAndCache(policy: component.policy)
+        return AnyPolicy(cachedPolicy)
+    }
+}
```

Comment:
Caching is only required at the top level and is done by `PolicyBuilder` already. `OneOfPolicy` will only be used inside a `PolicyBuilder` so we can remove this.